ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -230,6 +230,14 @@ void addTableFunction(final FunctionCall functionCall) {
     return tableFunctions;
   }
 
+  private static LogicalSchema buildStreamsSchema(final AliasedDataSource s) {
+    // Include metadata & key columns in the value schema to match the schema the streams
+    // topology will use.
+    return s.getDataSource()
+        .getSchema()
+        .withMetaAndKeyColsInValue(s.getDataSource().getKsqlTopic().getKeyFormat().isWindowed());
+  }
```

Comment:
Need to do this to ensure the Analyser has access to window bounds columns.  (Again, yes, this is a bit hacky).